src/main/scala/swoop/modelo/Bob.scala

```diff
@@ -0,0 +1,29 @@
+package swoop.modelo
+
+import scala.language.dynamics
+
+case class ModeloBobException(smth: String)  extends Exception(smth)
+
+case class Bob(
+    inputParams: Map[String, List[String]] = Map.empty[String, List[String]],
+    required: Set[String] = Set.empty[String],
+    requireAtLeastOne: Set[String] = Set.empty[String]
+  ) extends Dynamic {
+
+  def applyDynamic(name: String)(args: String*): Bob = {
+    copy(inputParams = inputParams ++ Map(name -> args.toList))
```

Comment:
I've seen this `copy(...)` pattern a few times in Scala now, but I don't understand what it's for. Why do you need to call `copy()` here?

Reply:
`copy` is how to make a new object and keep everything nice and immutable.